Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2209,6 +2214,44 @@ namespace Babylon
                 }
             }
 
+            Napi::Value FillPoses(const Napi::CallbackInfo& info)
+            {
+                const auto spaces = info[0].As<Napi::Array>();
+                auto transforms = info[2].As<Napi::Float32Array>();
+                if (spaces.Length() != transforms.ElementLength() >> 4)
+                {
+                    throw std::runtime_error{"Number of spaces doesn't match number of transforms * 16."};
+                }
+
+                for (uint32_t spaceIdx = 0; spaceIdx < spaces.Length(); spaceIdx++)
+                {
+                    const auto& jointSpace = *spaces[spaceIdx].As<Napi::External<xr::System::Session::Frame::Space>>().Data();
+                    const auto transformMatrix = CreateTransformMatrix(jointSpace, false);
+                    memcpy(transforms.Data() + (spaceIdx << 4), transformMatrix.data(), 4 /*Float32*/ * 16);
+                }
+
+                return Napi::Value::From(info.Env(), true);
+            }
+
+            Napi::Value FillJointRadii(const Napi::CallbackInfo& info) 
+            {
+                const auto spaces = info[0].As<Napi::Array>();
+                auto radii = info[1].As<Napi::Float32Array>();
+                if (spaces.Length() != radii.ElementLength()) 
+                {
+                    throw std::runtime_error{"Number of spaces doesn't match number of radii."};
+                }
+
+                for (uint32_t spaceIdx = 0; spaceIdx < spaces.Length(); spaceIdx++)
+                {
+                    const auto& jointSpace = *spaces[spaceIdx].As<Napi::External<xr::System::Session::Frame::JointSpace>>().Data();
+                    const auto jointRadius = jointSpace.PoseRadius;
+                    *(radii.Data() + spaceIdx) = jointRadius;
```

Comment:
This looks a lot less sketchy if you use array syntax: `radii.Data()[spaceIdx] = jointRadius;`